### PR TITLE
fix: Fix blank workflow details page after workflow submission

### PR DESCRIPTION
**Pull request by terrytangyuan**
Fixes https://github.com/argoproj/argo-workflows/issues/9371.

Signed-off-by: Yuan Tang <terrytangyuan@gmail.com>

### CODE REVIEW

**Review by jessesuen — APPROVED** [review]

File: ui/src/app/shared/conditions-panel.tsx [hunk 1]
```diff
@@ -24,7 +24,7 @@ export function hasWarningConditionBadge(conditions: Condition[]): boolean {
 
 export function hasArtifactGCError(conditions: Condition[]): boolean {
     for (const condition of conditions) {
-        if (condition.type === 'ArtifactGCError') {
+        if (condition && condition.type === 'ArtifactGCError') {
```
Comment (alexec) [inline ui/src/app/shared/conditions-panel.tsx:27]: minor - but could be `condition?.type`

File: ui/src/app/workflows/components/workflow-details/workflow-details.tsx [hunk 2]
```diff
@@ -276,7 +276,7 @@ export const WorkflowDetails = ({history, location, match}: RouteComponentProps<
                 if (e.type === 'DELETED') {
                     setError(new Error('Workflow deleted'));
                 } else {
-                    if (hasArtifactGCError(e.object.status.conditions)) {
+                    if (e.object.status.conditions && hasArtifactGCError(e.object.status.conditions)) {
```
Comment (alexec) [inline ui/src/app/workflows/components/workflow-details/workflow-details.tsx:279]: minor  `hasArtifactGCError` could tolerant null

**Review by alexec — APPROVED** [review]

**Comment by terrytangyuan** [comment]
@alexec Thanks for the suggestions!

File: ui/src/app/shared/conditions-panel.tsx [hunk 3]
```diff
@@ -23,8 +23,11 @@ export function hasWarningConditionBadge(conditions: Condition[]): boolean {
 }
 
 export function hasArtifactGCError(conditions: Condition[]): boolean {
+    if (conditions === undefined) {
```
Comment (alexec) [inline ui/src/app/shared/conditions-panel.tsx]: or `if (!conditions)

Comment (terrytangyuan) [reply ui/src/app/shared/conditions-panel.tsx]: Yep you are reviewing an outdated diff.

**Review by dpadhiar — APPROVED** [review]
LGTM

**Pull request merged**